frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/date_cell/date_editor.dart

```diff
@@ -100,55 +100,32 @@ class _CellCalendarWidgetState extends State<_CellCalendarWidget> {
 
   @override
   Widget build(BuildContext context) {
+    final List<Widget> children = [
+      StartTextField(popoverMutex: popoverMutex),
+      EndTextField(popoverMutex: popoverMutex),
+      const DatePicker(),
+      const TypeOptionSeparator(spacing: 12.0),
+      const EndTimeButton(),
+      const VSpace(4.0),
+      const _IncludeTimeButton(),
+      const TypeOptionSeparator(spacing: 8.0),
+      DateTypeOptionButton(popoverMutex: popoverMutex),
+      const VSpace(4.0),
+      const ClearDateButton(),
+    ];
+
     return BlocProvider(
       create: (context) => DateCellCalendarBloc(
         dateTypeOptionPB: widget.dateTypeOptionPB,
         cellData: widget.cellContext.getCellData(),
         cellController: widget.cellContext,
       )..add(const DateCellCalendarEvent.initial()),
-      child: BlocBuilder<DateCellCalendarBloc, DateCellCalendarState>(
-        builder: (context, state) {
-          final List<Widget> children = [
-            AnimatedSwitcher(
-              duration: const Duration(milliseconds: 300),
-              child: state.includeTime
-                  ? _TimeTextField(
-                      isEndTime: false,
-                      timeStr: state.time,
-                      popoverMutex: popoverMutex,
-                    )
-                  : const SizedBox.shrink(),
-            ),
-            if (state.includeTime && state.isRange) const VSpace(8.0),
-            AnimatedSwitcher(
-              duration: const Duration(milliseconds: 300),
-              child: state.includeTime && state.isRange
-                  ? _TimeTextField(
-                      isEndTime: true,
-                      timeStr: state.endTime,
-                      popoverMutex: popoverMutex,
-                    )
-                  : const SizedBox.shrink(),
-            ),
-            const DatePicker(),
-            const TypeOptionSeparator(spacing: 12.0),
-            const EndTimeButton(),
-            const VSpace(4.0),
-            const _IncludeTimeButton(),
-            const TypeOptionSeparator(spacing: 8.0),
-            DateTypeOptionButton(popoverMutex: popoverMutex),
-            const VSpace(4.0),
-            const ClearDateButton(),
-          ];
-
-          return ListView.builder(
-            shrinkWrap: true,
-            controller: ScrollController(),
-            itemCount: children.length,
-            itemBuilder: (BuildContext context, int index) => children[index],
-            padding: const EdgeInsets.only(top: 18.0, bottom: 12.0),
-          );
-        },
+      child: ListView.builder(
+        shrinkWrap: true,
+        controller: ScrollController(),
+        itemCount: children.length,
+        itemBuilder: (BuildContext context, int index) => children[index],
+        padding: const EdgeInsets.only(top: 18.0, bottom: 12.0),
```

Comment:
What are there differences between using the Column widget and using ListView.builder when shrinkWrap is set to true here?

Reply:
Performance wise, nothing. they both need to evaluate everything. When I wrote this a long time ago, I think the min height on the window was a bit lower, so I had to make sure it wouldn't overflow. But now it doesn't matter since it won't overflow at all. Should I change to Column?

Reply:
yes. it would be better to use the column here since the shrinkwrap is set to true and the widget here isn't scrollable.